### PR TITLE
drivers/io1_xplained: improve driver and add support for the sdcard and light sensor

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

After working on #8960, I realized that some things could be improved in the driver:
- instead of wrapping the on-board device API, one could directly access the device via their driver API (this is the case for the temperature sensor which uses at30tse75x)
- use finer DEBUG instead of global LOG_ functions
- add support for sdcard
- add support for light sensor

This PR does that.

### Issues/PRs references

None

### CODE REVIEW

**Comment by miri64** [comment]
What's with the light sensor on the extension?

File: tests/driver_io1_xplained/main.c [hunk 1]
```diff
@@ -21,16 +21,37 @@
...
 
-#define SLEEP_1S   (1 * 1000 * 1000u) /* 1 seconds delay between each test */
+#include "periph/gpio.h"
+
+#include "at30tse75x.h"
+#include "sdcard_spi.h"
+#include "io1_xplained.h"
+#include "io1_xplained_params.h"
+
+#define DELAY_1S   (1U) /* 1 seconds delay between each test */
```
Comment (miri64) [inline tests/driver_io1_xplained/main.c]: Why this rename?

**Comment by aabadie** [comment]
> What's with the light sensor on the extension?

Done: a3203ea. There might be some calibration required but I can't find useful information on this.

**Comment by miri64** [comment]
Just tested (mind you: I don't have an SD card plugged-in, because I currently have none at hand): The test get's stuck in the SD card initialization routine.

File: tests/driver_io1_xplained/main.c (hunk 1 above)
Comment (aabadie) [reply tests/driver_io1_xplained/main.c]: > Why this rename?

The comment is stating a delay, so I made it more in line with the comment

**Comment by aabadie** [comment]
> The test get's stuck in the SD card initialization routine.

I can adapt this using the Card detect pin but I think this is an issue with the underlying driver that should handle this (maybe I'm wrong).

**Comment by miri64** [comment]
> If you are using the extension I left during FOSDEM, it should have one :) (or maybe...)

I was not @ FOSDEM. I'm using the one I have in my drawer for years ;-).

**Comment by aabadie** [comment]
@miri64, I added the use of the card detect pin to skip the sdcard initialization and read if no sd card is plugged. You should be able to get the light level value.

File: drivers/io1_xplained/io1_xplained.c [hunk 2]
```diff
@@ -36,54 +41,73 @@ int io1_xplained_init(io1_xplained_t *dev, const io1_xplained_params_t *params)
...
     return IO1_XPLAINED_OK;
 }
 
-int io1_xplained_read_temperature(const io1_xplained_t *dev, float *temperature)
+int io1_xplained_read_light_level(uint16_t *light)
 {
-    if (at30tse75x_get_temperature(&dev->temp, temperature) < 0) {
-        LOG_ERROR("Cannot read temperature sensor.\n");
+    int sample = adc_sample(IO1_LIGHT_ADC_LINE, IO1_LIGHT_ADC_RES);
+
```
Comment (miri64) [inline drivers/io1_xplained/io1_xplained.c]: Trailing whitespaces introduced in this line ;-).

**Comment by aabadie** [comment]
Fixed the trailing whitespace, directly squashed

**Comment by aabadie** [comment]
It's all green for quite some time now, let's merge.

**Pull request merged**